js/src/evaluate.js

```diff
@@ -3,49 +3,74 @@
  * @author Nicholas C. Zakas
  */
 
+//-----------------------------------------------------------------------------
+// Typedefs
+//-----------------------------------------------------------------------------
+
+/** @typedef {import("./momoa").MomoaNode} MomoaNode */
+/** @typedef {import("./momoa").MomoaNodeParts} MomoaNodeParts */
+/** @typedef {import("./momoa").MomoaDocumentNode} MomoaDocumentNode */
+/** @typedef {import("./momoa").MomoaStringNode} MomoaStringNode */
+/** @typedef {import("./momoa").MomoaNumberNode} MomoaNumberNode */
+/** @typedef {import("./momoa").MomoaBooleanNode} MomoaBooleanNode */
+/** @typedef {import("./momoa").MomoaMemberNode} MomoaMemberNode */
+/** @typedef {import("./momoa").MomoaObjectNode} MomoaObjectNode */
+/** @typedef {import("./momoa").MomoaElementNode} MomoaElementNode */
+/** @typedef {import("./momoa").MomoaArrayNode} MomoaArrayNode */
+
 //-----------------------------------------------------------------------------
 // Exports
 //-----------------------------------------------------------------------------
 
 /**
  * Evaluates a Momoa AST node into a JavaScript value.
- * @param {Node} node The node to interpet.
+ * @param {MomoaNode} node The node to interpet.
  * @returns {*} The JavaScript value for the node. 
```

Comment:
Similar to my other comment, `{*}` is turned into `any` in TS, which is a type that leads to all kinds of issues. You can have a type like this:
```ts
type JSONValue =
	| Array<JSONValue>
	| boolean
	| number
	| string
	| { [property: string]: JSONValue }
	| null;
```
And use that instead 😄